### PR TITLE
Support big job workspace

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

### Description

- Save workspace without loading data into memory. 
- Load job data directly to file system when downloading to admin
- Improve download_job command with binary protocol to stream job files chunk by chunk, without loading into memory
- Unzip received files on admin client
- Fixed a bug in client_runner in sync runner processing

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/server/job_cmds.py [hunk 1]
```diff
@@ -613,25 +631,33 @@ def _unzip_data(self, download_dir, job_data, job_id):
...
+        with engine.new_context() as fl_ctx:
+            job_def_manager.get_storage_for_download(job_id, download_dir, DATA, JOB_ZIP, fl_ctx)
+            job_def_manager.get_storage_for_download(job_id, download_dir, META, META_JSON, fl_ctx)
+            job_def_manager.get_storage_for_download(job_id, download_dir, WORKSPACE, WORKSPACE_ZIP, fl_ctx)
 
-        except Exception as e:
-            conn.append_error(f"Exception occurred trying to get job from store: {secure_format_exception(e)}")
-            return
+            self.download_folder(
+                conn, job_id, download_file_cmd_name=AdminCommandNames.DOWNLOAD_JOB_FILE, control_id=job_id
```
Comment (yhwen) [inline nvflare/private/fed/server/job_cmds.py:662]: The "self.download_folder()" will try to download all the files within the folder. Before the job run completes, the "workspace" binary file does not exist in the job store. But we still create the "wokspace.zip" symlink to that file. The advantage is whenever the "workspace" binary file generated in the job store, then it becomes available automatically. However, before the job run completes, the download_job will get an error for downloading this workspace.zip file, becuase it does not exist. Should we handle this scenario in the download_folder()?

Alternately, we may consider not to create the "workspace.zip" symlink before the job run workspace has been created. But we need to handle one way or the other.

**Review by yhwen — CHANGES_REQUESTED** [review]

File: nvflare/private/fed/server/job_cmds.py (hunk 1 above)
Comment (yanchengnv) [reply nvflare/private/fed/server/job_cmds.py:661]: Right. Let's use the alternative approach: if workspace doesn't exist yet, don't create the symlink.

**Review by yhwen — APPROVED** [review]

File: nvflare/fuel/hci/chunk.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+HEADER_LEN = HEADER_STRUCT.size
+
+MARKER_DATA = 101
+MARKER_END = 102
+
+MAX_CHUNK_SIZE = 1024 * 1024
+
+
+def get_slice(buf, start: int, length: int):
+    return buf[start : start + length]
```
Comment (nvidianz) [inline nvflare/fuel/hci/chunk.py:28]: Use memoryview here could improve performance a little.

Comment (IsaacYangSLA) [reply nvflare/fuel/hci/chunk.py:28]: I think it's like a 'preference' or 'cosmetic issue.'

**Review by nvidianz — APPROVED** [review]

**Pull request merged**